api/v1alpha5/conversion.go

```diff
@@ -437,3 +452,21 @@ func Convert_v1alpha5_OpenStackClusterStatus_To_v1alpha8_OpenStackClusterStatus(
 func Convert_v1alpha8_OpenStackMachineSpec_To_v1alpha5_OpenStackMachineSpec(in *infrav1.OpenStackMachineSpec, out *OpenStackMachineSpec, s conversion.Scope) error {
 	return autoConvert_v1alpha8_OpenStackMachineSpec_To_v1alpha5_OpenStackMachineSpec(in, out, s)
 }
+
+func Convert_v1alpha5_OpenStackClusterSpec_To_v1alpha8_OpenStackClusterSpec(in *OpenStackClusterSpec, out *infrav1.OpenStackClusterSpec, s conversion.Scope) error {
+	err := autoConvert_v1alpha5_OpenStackClusterSpec_To_v1alpha8_OpenStackClusterSpec(in, out, s)
+	if err != nil {
+		return err
+	}
+
+	emptySubnet := SubnetFilter{}
+	if in.Subnet != emptySubnet {
```

Comment:
> (Not important) I've seen this pattern a couple of times now. What's the reason to write this rather than:
> 
> ```go
> if in.Subnet != SubnetFilter{} {
> ```
> 
> ?

The thing is probably that you need to use parentheses:

```go
if in.Subnet != (SubnetFilter{}) {
```

Reply:
Yup, parentheses would be needed